File: src/Frontend/Frontend/wwwroot/js/mm/node.js
class Node {
    constructor(title) {
        this.title = title;
        this.children = [];
        this.parent = null;
    }

    addChild(nodeTitle) {
        if (this.children.length >= globalConfig.MAX_CHILDREN_SIZE)
        {
            return;
        }
        const node = new Node(nodeTitle);
        node.parent = this;
        const pushed = this.children.push(node);
        return this.children[pushed - 1];
    }

    appendChild(node) {
        this.children.push(node);
        node.parent = this;
    }
}

File: ndk-compiler/src/main/java/compiler/ndk/mindMapTree/Node.java
package compiler.ndk.mindMapTree;

import java.util.ArrayList;
import java.util.List;

public class Node {

    private String title;

    private List<Node> children;

    public Node(String title) {
        this.title = title;
        this.children = new ArrayList<>();
    }

    public void appendChild(Node node) {
        this.children.add(node);
    }

}
